### PR TITLE
Add SPI abstraction in Rust

### PR DESCRIPTION
**Pull request by CohenArthur**
We need to rebase so that Skallwar's commit appears correctly.

### CODE REVIEW

File: rust/kernel/spi.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+               spi_driver: bindings::spi_driver
+               ) -> Self {
+        Registration {
+            this_module,
+            spi_driver
+        }
+    }
+
+    pub fn register(&mut self) -> KernelResult {
+        let res = unsafe { bindings::__spi_register_driver(self.this_module.0, &mut self.spi_driver as
```
Comment (kloenk) [inline rust/kernel/spi.rs:24]: Does __* need a lock?

File: rust/kernel/spi.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use crate::bindings;
+use crate::c_types;
+use crate::error::{Error, KernelResult};
+
+pub struct Registration {
+    this_module: &'static crate::ThisModule,
```
Comment (kloenk) [inline rust/kernel/spi.rs:8]: Does the module have to be stat of the registration?

**Comment by CohenArthur** [comment]
@Kloenk I am so sorry, I mistakenly opened this pull request. My goal was to make it against my fork. It is not ready for merging into this repository yet. My bad

**Comment by kloenk** [comment]
> @Kloenk I am so sorry, I mistakenly opened this pull request. My goal was to make it against my fork. It is not ready for merging into this repository yet. My bad

No worries. Maybe my input is also valid for that purpose :-)

**Pull request closed without merging**